src/lib/one-graph/client.js

```diff
@@ -0,0 +1,835 @@
+const os = require('os')
+
+const dotProp = require('dot-prop')
+const { buildClientSchema, parse } = require('graphql')
+const fetch = require('node-fetch')
+
+const { NETLIFYDEVLOG, NETLIFYDEVWARN, chalk, log } = require('../../utils')
+
+const {
+  defaultExampleOperationsDoc,
+  extractFunctionsFromOperationDoc,
+  generateFunctionsFile,
+  generateHandler,
+  readGraphQLOperationsSourceFile,
+  writeGraphQLOperationsSourceFile,
+  writeGraphQLSchemaFile,
+} = require('./netlify-graph')
+
+const ONEDASH_APP_ID = '0b066ba6-ed39-4db8-a497-ba0be34d5b2a'
+
+const httpOkLow = 200
+const httpOkHigh = 299
+
+const basicPost = async (url, options) => {
+  const reqBody = options.body || ''
+  const userHeaders = options.headers || {}
+
+  const headers = {
+    ...userHeaders,
+    'Content-Type': 'application/json',
+    'Content-Length': reqBody.length,
+  }
+
+  const timeoutMilliseconds = 30_000
+
+  const resp = await fetch(url, {
+    method: 'POST',
+    headers,
+    timeout: timeoutMilliseconds,
+    compress: true,
+    body: reqBody,
+  })
+
+  const respBody = await resp.text()
+
+  if (resp.status < httpOkLow || resp.status > httpOkHigh) {
+    console.warn('Response:', respBody)
+    throw new Error(`Netlify OneGraph return non - OK HTTP status code: ${resp.status}`)
+  }
+
+  return respBody
+}
+
+const fetchOneGraphSchemaJson = async (appId, enabledServices) => {
+  const url = `https://serve.onegraph.com/schema?app_id=${appId}&services=${enabledServices.join(',')}`
+  const headers = {}
+
+  try {
+    const response = await basicPost(url, {
+      method: 'GET',
+      headers,
+      body: null,
+    })
+
+    return JSON.parse(response)
+  } catch (error) {
+    console.error('Error fetching schema:', error)
+  }
+}
+
+const fetchOneGraphSchema = async (appId, enabledServices) => {
+  const result = await fetchOneGraphSchemaJson(appId, enabledServices)
+  const schema = buildClientSchema(result.data)
+  return schema
+}
+
+const fetchOneGraph = async (
+  accessToken,
+  appId,
+  query,
+  operationName,
+  variables,
+  // eslint-disable-next-line max-params
+) => {
+  const payload = {
+    query,
+    variables,
+    operationName,
+  }
+
+  const body = JSON.stringify(payload)
+  try {
+    const result = await basicPost(`https://serve.onegraph.com/graphql?app_id=${appId}&show_metrics=false`, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
+        Authorization: accessToken ? `Bearer ${accessToken}` : '',
+      },
+      body,
+    })
+
+    // @ts-ignore
+    const value = JSON.parse(result)
+    if (value.errors) {
+      console.warn(`${NETLIFYDEVWARN} fetchOneGraph errors`, operationName, JSON.stringify(value, null, 2))
+    }
+    return value
+  } catch {
+    return {}
+  }
+}
+
+const fetchOneGraphPersisted = async (
+  appId,
+  accessToken,
+  docId,
+  operationName,
+  variables,
+  // eslint-disable-next-line max-params
```

Comment:
- [ ] please use config object like mentioned above instead of disabling the lint rule